modules/tensor_mechanics/src/actions/CohesiveZoneActionBase.C

```diff
@@ -89,10 +90,15 @@ CohesiveZoneActionBase::validParams()
 
 CohesiveZoneActionBase::CohesiveZoneActionBase(const InputParameters & params) : Action(params)
 {
+  // FIXME: suggest to use action of action to add this to avoid changing the input parameters in
+  // the warehouse.
+  const auto & parameters = _app.getInputParameterWarehouse().getInputParameters();
+  InputParameters & pars(*(parameters.find(uniqueActionName())->second.get()));
+
   // check if a container block with common parameters is found
   auto action = _awh.getActions<CommonCohesiveZoneAction>();
   if (action.size() == 1)
-    _pars.applyParameters(action[0]->parameters());
```

Comment:
Now the parameter is held in the warehouse and the action only owns a const reference of it. To get access to that input parameters, we have to inquire the warehouse. I feel you can use an action of action to replace the current syntax used by this action and to construct the full input parameters and then let the action factory create this action.

Reply:
Seems like a duplicate thread of https://github.com/idaholab/moose/pull/22069/files#r968702615